### ISSUE TITLE
SA1008 doesn't support params of tuples

### ISSUE THREAD

**Issue by nilobject** [DotNetAnalyzers/StyleCopAnalyzers#2568]
I was feeling adventurous this morning and wrote a method definition like this:

`public void SomeFunction(params (string name, string value)[] options)`

I got this warning:

`warning SA1008: Opening parenthesis should not be preceded by a space. `

It seems like this usage of params with tuples should not be warned against, because `params(string name, string value)` looks wrong -- it makes params look like a function call. This is of course just my opinion.

Removing the space does allow the code to be analyzed without warning, and it does compile still, so this is a minor issue at best (if it's even considered bad style in the first place).

**Comment by FelschR** [comment]
I#m having the same issue when creating a Dictionary with nested tuple:

```csharp
var extendedTimes = new Dictionary<DayOfWeek, (string start, string end)>
```

This results in both a `SA1008` (start) & `SA1009` (end).

Following the guideline here would be very strange:
```csharp
var extendedTimes = new Dictionary<DayOfWeek,(string start, string end) >
```

**Comment by negativeeddy** [comment]
same here with simple tuple assignment,, although you can't fix it because if you remove the space before the parenthesis, you get the SA1003 '=' must be followed by a space
`var tmp = (1, "2");`

**Comment by vweijsters** [comment]
@FelschR and @negativeeddy: I cannot reproduce the problems that you describe using version 1.1.0-beta004. Did you use this version when then issue occurred?

**Comment by vweijsters** [comment]
@nilobject This issue has been fixed in #2478, which will be available when 1.1.0-beta005 is released
